contracts/JBPaymentTerminalStore.sol

```diff
@@ -256,31 +254,30 @@ abstract contract JBPaymentTerminalStore is IJBPaymentTerminalStore {
     The msg.sender must be an IJBPaymentTerminal. The amount specified in the params is in terms of the msg.senders tokens.
 
     @param _payer The original address that sent the payment to the terminal.
-    @param _amount The amount of terminal tokens being paid, as a fixed point number with 18 decimals.
+    @param _amount The amount of tokens being paid, as a fixed point number. Includes the token being paid, the value, the number of decimals included, and the currency of the amount.
     @param _projectId The ID of the project being paid.
     @param _beneficiary The address that should receive benefits from the payment.
+    @param _baseWeightCurrency The currency to base token issuance on.
     @param _memo A memo to pass along to the emitted event, and passed along the the funding cycle's data source and delegate.
     @param _metadata Bytes to send along to the data source, if one is provided.
 
     @return fundingCycle The project's funding cycle during which payment was made.
-    @return weight The weight according to which new token supply was minted, as a fixed point number with 18 decimals.
     @return tokenCount The number of project tokens that were minted, as a fixed point number with 18 decimals.
     @return delegate A delegate contract to use for subsequent calls.
     @return memo A memo that should be passed along to the emitted event.
   */
   function recordPaymentFrom(
     address _payer,
-    uint256 _amount,
+    JBTokenAmount memory _amount,
```

Comment:
calldata instead of memory (should be -2k)

Reply:
Hmm, I'll go hunt for these actually, we might save additional gas with it

Reply:
oo im real curious